spec/app/domain/authentication/authn-jwt/signing_key/fetch_signing_key_settings_from_variables_spec.rb

```diff
@@ -0,0 +1,219 @@
+                                "CONJ00086E Signing key URI configuration is invalid")
+      end
+    end
+
+    context "'jwks-uri' exits and 'provider-uri' do not exists" do
+      context "fetching 'jwks-uri' successfully" do
+        subject do
+          ::Authentication::AuthnJwt::SigningKey::FetchSigningKeySettingsFromVariables.new(
+            check_authenticator_secret_exists:  mocked_check_authenticator_secret_exists_jwks,
+            fetch_authenticator_secrets: mocked_fetch_authenticator_secrets_exist_jwks,
+            logger: mocked_logger
+          ).call(
+            authenticator_input: mocked_authenticator_input
+          )
+        end
+
+        it "equals to expected signing key settings" do
+          expect(subject.uri).to eql(mocked_jwks_uri)
+          expect(subject.type).to eql(mocked_jwks_type)
+        end
+      end
+
+      context "fetching 'jwks-uri' not successfully" do
+        subject do
+          ::Authentication::AuthnJwt::SigningKey::FetchSigningKeySettingsFromVariables.new(
+            check_authenticator_secret_exists:  mocked_check_authenticator_secret_exists_jwks,
+            fetch_authenticator_secrets: mocked_fetch_authenticator_secrets_not_exist_jwks,
+            logger: mocked_logger
+          ).call(
+            authenticator_input: mocked_authenticator_input
+          )
+        end
+
+        it "raise an error" do
+          expect { subject }.to raise_error(
+                                  "mockedRequiredSecretMissingError")
+        end
+      end
+    end
+
+    context "'jwks-uri' does not exist and 'provider-uri' exists" do
+      context "fetching 'provider-uri' successfully" do
+        subject do
+          ::Authentication::AuthnJwt::SigningKey::FetchSigningKeySettingsFromVariables.new(
+            check_authenticator_secret_exists:  mocked_check_authenticator_secret_exists_provider,
+            fetch_authenticator_secrets: mocked_fetch_authenticator_secrets_exist_provider,
+            logger: mocked_logger,
+            ).call(
+            authenticator_input: mocked_authenticator_input
+          )
+        end
+
+        it "equals to expected signing key settings" do
+          expect(subject.uri).to eql(mocked_provider_uri)
+          expect(subject.type).to eql(mocked_provider_type)
+        end
+      end
+
+      context "fetching 'provider-uri' not successfully" do
+        subject do
+          ::Authentication::AuthnJwt::SigningKey::FetchSigningKeySettingsFromVariables.new(
+            check_authenticator_secret_exists:  mocked_check_authenticator_secret_exists_provider,
+            fetch_authenticator_secrets: mocked_fetch_authenticator_secrets_empty_provider,
+            logger: mocked_logger
+          ).call(
+            authenticator_input: mocked_authenticator_input
+          )
+        end
+
+        it "raise an error" do
+          expect { subject }.to raise_error(
+                                  "mockedRequiredSecretMissingError")
```

Comment:
```suggestion
          expect { subject }.to raise_error(mocked_required_secret_missing_error)
```
Please use a variable name instead of the value